extensions/barrage/src/main/java/io/deephaven/extensions/barrage/chunk/SingleElementListHeaderReader.java

```diff
@@ -0,0 +1,58 @@
+//
+// Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.extensions.barrage.chunk;
+
+import io.deephaven.chunk.WritableChunk;
+import io.deephaven.chunk.attributes.Values;
+import io.deephaven.util.datastructures.LongSizedDataStructure;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.io.DataInput;
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.PrimitiveIterator;
+
+public class SingleElementListHeaderReader<READ_CHUNK_TYPE extends WritableChunk<Values>>
```

Comment:
Correct; this is only for columns as a list feature. (Since otherwise our mod column vectors all have different sizes which cause some of the clients to crash/segfault/error/etc.) I'll add some public top level javadoc.